src/tfs/tfs.c

```diff
@@ -603,14 +608,16 @@ static fs_status update_extent_length(fsfile f, extent ex, u64 new_length)
     return FS_STATUS_OK;
 }
 
-static u64 extend(fsfile f, extent ex, sg_list sg, range blocks, merge m)
+static u64 extend(fsfile f, extent ex, sg_list sg, range blocks, merge m, boolean write)
 {
     u64 free = ex->allocated - range_span(ex->node.r);
     range r = irangel(ex->node.r.end, free);
     range i = range_intersection(r, blocks);
     tfs_debug("   %s: node %R, free 0x%lx (%R), i %R\n", __func__, ex->node.r, free, r, i);
     if (range_span(i) == 0)
         return blocks.start;
+    if (!write)
```

Comment:
In fill_gap, the extent meta is updated even if write is false. In a similar vein, shouldn't this include the call to update_extent_length even if write is false?

Reply:
Yes, I think you are right.